scheduler/orchestrator/anomaly_detector.py

```
@@ -238,8 +237,8 @@ def __init__(
     
     def list_s3_directories(self, hub: str, stage: str) -> list[str]:
         """List all directories in S3 for a hub/stage, sorted by date descending."""
-        cmd = f"aws s3 ls s3://{self.s3_bucket}/{hub}/{stage}/ --profile {self.aws_profile}"
-        result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
+        cmd = ["aws", "s3", "ls", f"s3://{self.s3_bucket}/{hub}/{stage}/", "--profile", self.aws_profile]
+        result = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)